src/PatchOrchestrationApplication/CoordinatorService/src/RepairManagerHelper.cs

```diff
@@ -274,6 +279,156 @@ internal async Task PrintRepairTasks(CancellationToken cancellationToken)
+                RepairTaskList processingTaskList = await this.GetRepairTasksUnderProcessing(cancellationToken);
+                cancellationToken.ThrowIfCancellationRequested();
+                if (claimedTaskList.Any())
+                {
+                    if (!processingTaskList.Any())
+                    {
+                        // This means that repair tasks are not getting approved.
+                        ClusterHealth clusterHealth = await this.fabricClient.HealthManager.GetClusterHealthAsync();
+                        if (clusterHealth.AggregatedHealthState == HealthState.Error)
+                        { 
+                            // Reset Count
+                            postUpdateCount = 0;
+                            string warningDescription = "Cluster is unhealthy. POA Repair task created for OS update will not be approved. Please take cluster to healthy state for POA to start working.";
+                            HealthManagerHelper.PostNodeHealthReport(this.fabricClient, this.context.ServiceName, RMTaskUpdateProperty, warningDescription, HealthState.Ok, 5);
+                        }
+                        else
+                        {
+                            postUpdateCount++;
+                            if (postUpdateCount > 60)
+                            {
+                                // Reset Count and throw a warning on the service saying we dont know the reason. But POA not is not approving tasks.
+                                postUpdateCount = 0;
+                                string warningDescription = "POA repair tasks are not getting approved. So, update installation is halted. Please try to find out why is this blocked.";
+                                HealthManagerHelper.PostNodeHealthReport(this.fabricClient, this.context.ServiceName, RMTaskUpdateProperty, warningDescription, HealthState.Warning, 90);
+                            }
+                        }
+                    }
+                    else
+                    {
+                        // Reset Count
+                        postUpdateCount = 0;
+                        await PostRMTaskNodeUpdate(cancellationToken);
+                    }
+                }
+                else
+                {
+                    // Reset Count
+                    postUpdateCount = 0;
+                    if (processingTaskList.Any())
+                    {
+                        await PostRMTaskNodeUpdate(cancellationToken);
+                    }
+                    else
+                    {
+                        // Post the health event saying that there is no repair task and things are working fine.
+                        string description = "No claimed tasks and no processing tasks are found.";
+                        HealthManagerHelper.PostNodeHealthReport(this.fabricClient, this.context.ServiceName, RMTaskUpdateProperty, description, HealthState.Ok, -1);
+                    }
```

Comment:
Are you using this to clear the health report? Ideally you should post the healthy with the time limit in which the outer loop runs and based on that repost the event before the previous one expires.

Reply:
Yes..either of the above will get posted on the service